Multivariate linear regression using sklearn

In [3]:
import pandas
from sklearn import linear_model
#data
df= pandas.read_excel("car.xlsx")
X = df[['Weight', 'Volume']]
y = df['CO2']
#model
model= linear_model.LinearRegression()
model.fit(X, y)
#prediction
predictedCO2 = model.predict([[2300, 1300]])
print(predictedCO2)
print(model.coef_)

[107.2087328]
[0.00755095 0.00780526]


We have already predicted that if a car with a 1300cm3 engine weighs
2300kg, the CO2 emission will be approximately 107g.

•These values tell us that if the weight increase by 1kg, the CO2
emission increases by 0.00755095g.

•And if the engine size (Volume) increases by 1 cm3, the CO2 emission
increases by 0.00780526 g